### DISCUSSION TITLE
Service auto discovery for non GKE workloads

### DISCUSSION THREAD

**Discussion by pandrian** [GoogleCloudPlatform/prometheus-engine#430] [Q&A] [answered]
Hi, 
is it possible to scrape external endpoints (gce_sd_configs) using service auto discovery with GMP? and if so are there any examples around to have a look? 

Thanks!

**Comment by lyanco** [comment, marked as answer]
GMP's managed collection really only works for k8s and k8s workloads. For VMs, we recommend either of the following:

[Self-deployed collection](https://cloud.google.com/stackdriver/docs/managed-prometheus/setup-unmanaged), which you can deploy using prom-operator or something and configure to find VMs using gce_sd_configs
[The Ops Agent](https://cloud.google.com/monitoring/agent/ops-agent/prometheus), which runs as an agent on VMs and scrapes colocated exporters, making gce_sd_config not needed any more.
